python/ray/data/block.py

```diff
@@ -154,22 +156,15 @@ class BlockMetadata:
         exec_stats: Execution stats for this block.
     """
 
-    def __init__(
-        self,
-        *,
-        num_rows: Optional[int],
-        size_bytes: Optional[int],
-        schema: Union[type, "pyarrow.lib.Schema"],
-        input_files: List[str],
-        exec_stats: Optional[BlockExecStats]
-    ):
-        if input_files is None:
-            input_files = []
-        self.num_rows: Optional[int] = num_rows
-        self.size_bytes: Optional[int] = size_bytes
-        self.schema: Optional[Any] = schema
-        self.input_files: List[str] = input_files
-        self.exec_stats: Optional[BlockExecStats] = exec_stats
+    num_rows: Optional[int]
+    size_bytes: Optional[int]
+    schema: Optional[Union[type, "pyarrow.lib.Schema"]]
```

Comment:
The type annotation for `schema` was previously `Union[type, "pyarrow.lib.Schema"]`. I've changed it to  `Optional[Union[type, "pyarrow.lib.Schema"]]` so that it's consistent with the docstring: 
> schema: The pyarrow schema or types of the block elements, or **None**.